### PR TITLE
chore(examples): update rust example to use action configs

### PR DESCRIPTION
**Pull request by Walther**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/garden-io/garden/issues/3740

**Special notes for your reviewer**:

First time converting an example to the new config format.
Is this correct and aligned with the best practices?

### CODE REVIEW

File: examples/rust/README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ Example of using the Rust programming language with Garden.
 Local development with hot reload:
```
Comment (vvagaytsev) [inline examples/rust/README.md:5]: The term "hot reload" can be confused with the old `--hot-reload` flag. Let's use the "code synchronization" term instead.

File: examples/rust/backend/backend.garden.yml [hunk 2]
```diff
@@ -1,27 +1,34 @@
...
-        containerPort: 3000
-    healthCheck:
-      httpGet:
-        path: /
-        port: http
-    ingresses:
-      - path: /
-        port: http
-        hostname: backend.${var.base-hostname}
+---
```
Comment (vvagaytsev) [inline examples/rust/backend/backend.garden.yml:8]: Nit: an empty line can be placed after the separator :)

Comment (Walther) [reply examples/rust/backend/backend.garden.yml:8]: Prettier seems to prefer not having an empty line after the separator. Do we want to change prettier configuration to prefer a newline after the yaml separator?

Comment (Walther) [reply examples/rust/backend/backend.garden.yml:8]: The yaml specification also seems to not have a newline after the separator in their examples: https://yaml.org/spec/1.2.2/#22-structures

Comment (vvagaytsev) [reply examples/rust/backend/backend.garden.yml:8]: Ah, ok :) Thanks for sharing this!

**Review by vvagaytsev — APPROVED** [review]
Looks good, great job! 👍 
If the example works fine with and without `--dev` flag, then it's ready to go! 🚀

**Comment by Walther** [comment]
Thanks for the speedy review!

**Pull request merged**